java/ql/src/experimental/Security/CWE/CWE-400/ThreadPauseSink.qll

```diff
@@ -0,0 +1,32 @@
+/** Provides sink models related to pausing thread operations. */
+
+import java
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.ExternalFlow
+
+/** `java.lang.Math` data model for value comparison in the new CSV format. */
+private class MathCompDataModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "java.lang;Math;false;min;;;Argument[0..1];ReturnValue;taint",
+        "java.lang;Math;false;max;;;Argument[0..1];ReturnValue;taint"
```

Comment:
These should have `value` as their `kind`.

```suggestion
        "java.lang;Math;false;min;;;Argument[0..1];ReturnValue;value",
        "java.lang;Math;false;max;;;Argument[0..1];ReturnValue;value"
```

Reply:
Thanks @atorralba for reviewing this PR. I've changed taint model to fluent model.

Reply:
May I ask why this should be `value` flow?
Why is `x = min(a, b)` `value` flow when either the value of `a` or `b` isn't preserved?

Reply:
Even if only one of the two arguments flows into the return value on each call, the one that does is preserving its value. With primitive types it gets a little fuzzier, but generally speaking we want value-preserving steps when it's the same exact instance of an object being passed around (e.g. the same `String` instance, not a copy of it), even if conditionally. You can always add another CSV row to model the other branch if necessary.

See for example how we modeled [`JSONArray.optInt`](https://javadoc.io/doc/org.json/json/20171018/org/json/JSONArray.html) which is a similar case (`defaultValue` will only be preserved if there's no value for `index`): https://github.com/github/codeql/blob/eda7616a23970c9a9ff8f0734b1c5971fe5b114d/java/ql/lib/semmle/code/java/frameworks/JsonJava.qll#L71